pkg/kademlia/config.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package kademlia
+
+import (
+	"context"
+	"net"
+
+	"github.com/zeebo/errs"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/provider"
+	proto "storj.io/storj/protos/overlay"
+)
+
+var (
+	// Error defines a Kademlia error
+	Error = errs.Class("kademlia error")
+	mon   = monkit.Package()
+)
+
+type ctxKey int
+
+const (
+	ctxKeyKad ctxKey = iota
+)
+
+// Config defines all of the things that are needed to start up Kademlia
+// server endpoints (and not necessarily client code).
+type Config struct {
+	BootstrapAddr string `help:"the kademlia node to bootstrap against" default:"bootstrap-dev.storj.io:8080"`
+	// TODO(jt): remove this! kademlia should just use the grpc server
+	TODOListenAddr string `help:"the host/port for kademlia to listen on. TODO(jt): this should be removed!" default:"127.0.0.1:7776"`
+}
+
+// Run implements provider.Responsibility
+func (c Config) Run(ctx context.Context, server *provider.Provider) (
+	err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	// TODO(jt): don't split the host/port
```

Comment:
because the Go standard library uses addresses almost uniformly. By standardizing on addresses, our code is more flexible, conforms with the Go standard library conventions, and prevents issues like people forgetting to allow the host to be configured and only the port. You see that a lot in our current codebase - you can only configure the port and not a host, thus not allowing the service to configure what network interface it listens on.

Reply:
👍